text/go/0001-go-bom.md

```diff
@@ -0,0 +1,163 @@
+capture this information in Go projects (as compared with other languages like nodejs and rust, where the package
+description files contain a field for this metadata). See unresolved questions below.
+
+In order to support this tool in both the online and offline cases, we will host the binary tool on
+the [dep-server](https://github.com/paketo-buildpacks/dep-server).
+
+### Performance
+
+The addition of this tool to the language family order groupings might have an impactful change to the overall
+performance of the buildpacks.
+
+Performance metrics are still TBD, but for a typical project (e.g. `pack`) we observe that it takes about 10-30 seconds
+to construct a BOM.
+
+We can investigate ways to improve this performance.
+
+### Language Family Additions
+
+Using the CycloneDX tool to generate the BOM will involve some changes to the Go language family buildpack order groups
+as follows:
+
+* A single BOM generation buildpack will be added. It's primary functions are:
+    * perform the BOM dependency tool retrieval from the dep-server,
+    * execute BOM generation with the tool (`cyclonedx-gomod app -o bom.json`), and
+    * contribute the BOM to the `launch.toml` `[bom]` label.
+
+* For the Go Mod buildpack order group, the BOM generation buildpack will run directly before the `go-build` buildpack.
+
+#### Detection
+
+The Go Module BOM Generator CNB always detects with the following contract:
+
+* Requires {`go-dist`} during `build`
+* Provides none
+
+#### Build
+
+The build phase will perform a few tasks as mentioned above.
+
+1. Perform the BOM dependency tool retrieval from the dep-server. It will get added to a build-time layer. It will
+   contribute a BOM entry for the tool itself on the `build.toml` `[bom]` section.
+
+2. Execute BOM generation with the installed tool via the `cyclonedx-gomod app -o bom.json` command in the root
+   directory of the application. There is an unresolved question around determining which main package to run against if
+   the application source contains multiple `main` packages.
+
+3. Unmarshal the CycloneDX JSON BOM and transform each BOM entry into a
+   [`postal.Dependency`](https://github.com/paketo-buildpacks/packit/blob/c5a40518f2c6bd913ade999b9e2d58d6892d2ea9/postal/buildpack.go#L12)
+   type. This step and step 5 will eventually go away if we support CycloneDX as a BOM format.
+
+4. (Optionally) generate `CPE` and `pURL` from the data and add it to the
+   `postal.Dependency`.
+
+5. Run the
+   [`postal.GenerateBillOfMaterials`](https://github.com/paketo-buildpacks/packit/blob/c5a40518f2c6bd913ade999b9e2d58d6892d2ea9/postal/service.go#L186)
+   command.
+
+6. Clean up the generated BOM file
+
+#### Potential Future Optimizations
+
+Given future user interest in a way to opt out of this buildpack beyond a custom order grouping, an optimization that
+could be made is the inclusion of a
+`BP_ENABLE_MODULE_BOM` environment variable that can be set during container build-time.
```

Comment:
Is this intentionally left out of scope for now? A penalty of 10-30 seconds per build seems steep without an off switch.